src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/CodeBlockService.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Razor;
+internal static class CodeBlockService
+{
+    /// <summary>
+    ///  Creates a <see cref="TextEdit"/> that will place the formatted generated method within a @code block in the file.
+    /// </summary>
+    /// <param name="code">The <see cref="RazorCodeDocument"/> of the file where the generated method will be placed.</param>
+    /// <param name="generatedMethod">
+    ///  The skeleton of the generated method where a '0' should be placed anywhere that needs to have some initial indenting, '1' for the inner indent of the method, and a '2' for where the method name should be placed.
+    ///  It should look something like: 0public void 2(){\r\n1throw new NotImplementedException();\r\n0}
+    /// </param>
+    /// <param name="methodName">The name of the method that is being generated.</param>
+    /// <returns>A <see cref="TextEdit"/> that will place the formatted generated method within a @code block in the file.</returns>
+    public static TextEdit CreateFormattedTextEdit(RazorCodeDocument code, string generatedMethod, string methodName)
+    {
+        var indentSize = code.GetCodeGenerationOptions().IndentSize;
```

Comment:
Sadly, this is not the right indent size. This is for the compiler when it generates the backing C# file, but we want the indent for the Razor file. That comes from `RazorLSPOptions`, which you get by injecting the `RazorLSPOptionsMonitor` service into your resolver, getting the current options from it, and passing them into here in some form.

Not your fault though, the method name does look like its exactly what you want :)